src/MCPServer/lib/linkTaskManagerChoice.py

```diff
@@ -138,9 +138,12 @@ def xmlify(self):
 
     @log_exceptions
     @auto_close_db
-    def proceedWithChoice(self, chain, agent, delayTimerStart=False):
-        if agent:
-            self.unit.setVariable("activeAgent", agent, None)
+    def proceedWithChoice(self, chain, user_id, delayTimerStart=False):
+        if user_id is not None:
+            agent_id = UserProfile.objects.get(user_id=int(user_id)).agent_id
```

Comment:
If we're doing this same thing in a few files, maybe make a helper function to return the `agent_id` as a string?